package exceptionhandling;

public class LearnExceptionHandling {
    //Exception: something is not regular outcomes
    //Five Key Words: try, catch, throw, throws, finally

    //Arithmetic Exception
    //It is thrown when an exceptional condition has occurred in arithmetic operation. ArrayIndexOutOfBoundsException
    //It is thrown to indicate that an array has been accessed with an illegal index. The Index is either negative or positive
    // ClassNotFoundException    //This Exception is raised when we try to access a class whose definition is not found:
    // FileNotFoundException
    //This exception is raised when a file is not accessible or does nto open:
    // IOException
    //It is thrown when thread is waiting, sleeping, or doing some processing, and it is interrupted:
    // NoSuchFieldException
    //It is thrown when a class does not contain the field or variable specified:
    // NoSuchMethodException
    //It is thrown when accessing a method which is not found:
    // NullPointerException
    //It is raised when referring to the members of a null object. Null represents nothing:
    // NumberFormatException
    //It is raised when a method could not convert a string into a numeric format:
    // RuntimeException
    //Represents any exception which occurs during runtime:
    //StringIndexOutOfBoundsException
    //It is thrown by String Class methods to indicate that an index is either negative than the size of the string



}
